unit/memleak.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import gc
+import tracemalloc
+
+import numpy as np
+
+
+def run_memleak_test(bench, iterations, report):
+    from matplotlib.cbook import report_memory
+
+    tracemalloc.start()
+
+    starti = min(10, iterations / 2)
+    endi = iterations
+
+    malloc_arr = np.empty((endi,), dtype=np.int64)
+    rss_arr = np.empty((endi,), dtype=np.int64)
+    nobjs_arr = np.empty((endi,), dtype=np.int64)
+    garbage_arr = np.empty((endi,), dtype=np.int64)
+
+    for i in range(endi):
+        bench()
+
+        gc.collect()
+        rss = report_memory()
+        malloc, peak = tracemalloc.get_traced_memory()
+        nobjs = len(gc.get_objects())
+        garbage = len(gc.garbage)
+        print("{0: 4d}: pymalloc {1: 10d}, rss {2: 10d}, nobjs {3: 10d}, garbage {4: 10d}".format(
+            i, malloc, rss, nobjs, garbage))
+
+        malloc_arr[i] = malloc
+        rss_arr[i] = rss
+        nobjs_arr[i] = nobjs
+        garbage_arr[i] = garbage
+
+    print('Average memory consumed per loop: %1.4f bytes\n' %
+          (np.sum(rss_arr[starti+1:] - rss_arr[starti:-1]) / float(endi - starti)))
```

Comment:
This looks like just the sum of the differences, which is the end value minus the start value.
```python
((rss_arr[-1] - rss_arr[starti]) / float(endi - starti))
```